src/backend/tcop/postgres.c

```diff
@@ -1589,25 +1591,30 @@ send_guc_to_QE(List *guc_list, bool is_restore)
 		}
 		PG_CATCH();
 		{
-			/* if some guc can not restore successful
-			 * we can not keep alive gang anymore.
-			 */
-			DisconnectAndDestroyAllGangs(true);
-			CheckForResetSession();
-			/*
-			 * when qe elog an error, qd will use ReThrowError to
-			 * re throw the error, the errordata_stack_depth will ++,
-			 * when we catch the error we should reset errordata_stack_depth
-			 * by FlushErrorState.
-			 */
-			FlushErrorState();
-			/*
-			 * this is a top-level catch block and we are responsible for
-			 * restoring the right memory context.
-			 */
+			if (!elog_dismiss(WARNING))
+				PG_RE_THROW();
+
+			is_success = false;
+
 			MemoryContextSwitchTo(oldcontext);
 		}
 		PG_END_TRY();
+
+		/*
+		 * Can't continue the transaction or keep the gang alive if GUCs are out
+		 * of sync. Explode!
+		 */
+		if (!is_success)
```

Comment:
https://github.com/arenadata/gpdb/pull/1416/files#r2028103896 and https://github.com/arenadata/gpdb/pull/1416/files#r2028106060